### PR TITLE
gh-107801: Improve the accuracy of os.lseek docs

### PR DESCRIPTION
**Pull request by erlend-aasland**
- name the last parameter *whence*, like it is for seek() methods on
  file objects
- add param docstrings
- structure the valid *whence* values as a list

* Issue: gh-107801

----
:books: Documentation preview :books:: https://cpython-previews--107935.org.readthedocs.build/

### CODE REVIEW

File: Modules/posixmodule.c
Comment (AA-Turner) [inline Modules/posixmodule.c]: Perhaps:
```suggestion
Set and return the position of a file descriptor.
```

File: Modules/posixmodule.c
Comment (AA-Turner) [inline Modules/posixmodule.c:10439]: I'm not sure this is needed, but perhaps explaining that 'position' is measured in bytes is useful?

File: Doc/library/os.rst
Comment (AA-Turner) [inline Doc/library/os.rst]:
```suggestion
   by *whence*, and return the new position in bytes relative to
```

**Review by AA-Turner — APPROVED** [review]

**Review by pitrou — APPROVED** [review]

File: Modules/posixmodule.c
Comment (erlend-aasland) [reply Modules/posixmodule.c]: I think this is good.

File: Modules/posixmodule.c
Comment (erlend-aasland) [reply Modules/posixmodule.c:10439]: I think it is ok, since this is a low-level API.

File: Modules/posixmodule.c
Comment (erlend-aasland) [reply Modules/posixmodule.c]: OTOH, the suggested change is very explicit and clear. Perhaps it makes for less confusion.

Comment (AA-Turner) [reply Modules/posixmodule.c]: No strong opinion, it was an attempt to reduce some duplication -- but just a cosmetic change rather than substance.

Comment (erlend-aasland) [reply Modules/posixmodule.c]: I'll ponder on it until tomorrow :)

**Review by serhiy-storchaka — COMMENTED** [review]
See my comment in https://github.com/python/cpython/pull/107936#pullrequestreview-1578033303. These changes are not independent and should be considered in one PR.

**Comment by erlend-aasland** [comment]
> See my comment in [#107936 (review)](https://github.com/python/cpython/pull/107936#pullrequestreview-1578033303). These changes are not independent and should be considered in one PR.

I think it is fine to document constants in one PR and update the docs of the function in another PR. Documenting the constants also unlock my draft PR to clean up the Sphinx warnings in Lib/library/os.rst.

File: Modules/posixmodule.c
Comment (erlend-aasland) [reply Modules/posixmodule.c]: I'll revert this line; the original is fine, and my suggestion neither improves nor worsens the situation.

Comment (erlend-aasland) [reply Modules/posixmodule.c]: Reverted in ca27fca6e967b2cc43d8ad465bc1354515c9fb8e

**Comment by erlend-aasland** [comment]
I'll hold this until #107936 has landed.

**Review by AA-Turner — APPROVED** [review]
Looks good!

A

**Pull request merged**
